src/stream/stream_backend.rs

```diff
@@ -370,6 +387,109 @@ fn create_redirect_stream(
     return Ok(StreamType::REDIRECT(stream));
 }
 
+fn create_webrtc_stream(
+    video_and_stream_information: &VideoAndStreamInformation,
+) -> Result<StreamType, SimpleError> {
+    let endpoints = &video_and_stream_information.stream_information.endpoints;
+    if endpoints.len() != 1 {
+        return Err(SimpleError::new("Multiple endpoints are not accepted."));
+    }
+    let endpoint = &endpoints[0];
+    let expected_endpoint = url::Url::parse("turn://user:pwd@0.0.0.0:3478".into()).unwrap();
+    if endpoint != &expected_endpoint {
+        return Err(SimpleError::new(format!(
+            "The only accepted endpoint is: {expected_endpoint}, but {endpoint} was given."
+        )));
+    }
+
+    let configuration = match &video_and_stream_information
+        .stream_information
+        .configuration
+    {
+        CaptureConfiguration::VIDEO(configuration) => configuration,
+        _ => return Err(SimpleError::new(format!(
+            "The only accepted configuration type is the VideoCaptureConfiguration, which is like {supported_configuration} ",
+            supported_configuration="VideoCaptureConfiguration { \"encode\": <string>, \"height\": <integer>, \"width\": <integer>, \"frame_interval\": <integer/integer> }",
```

Comment:
Isn't a better way of doing it ?

Reply:
Probably... but not for cheap. I'd asked about it in the BR rust community and it seems that the only solution to make this automatic would be to create a specific macro for that. I'd tried and it feels just too much effort for now... My approach was basically to create a macro that adds a custom implementation of a function that prints <name>: <type> of each field, but to use it, we would need to wrap all structs with that macro - that's ugly.

Reply:
It would be helpful that serde could provide that for us, but we can take a better look on the future.